stl/inc/type_traits

```diff
@@ -1151,11 +1151,17 @@ template <size_t _Len, class... _Types>
 using aligned_union_t = typename aligned_union<_Len, _Types...>::type;
 
 // STRUCT TEMPLATE underlying_type
+template <class _Ty, bool = is_enum_v<_Ty>>
+struct _Underlying_type {};
+
 template <class _Ty>
-struct underlying_type { // determine underlying type for enum
+struct _Underlying_type<_Ty, true> {

```

Comment:
Style request: When you have a `bool` like this, I recommend making the primary template the `true` case, and then provide a specialization for `false`. This makes the code more readable - the primary template mentions `is_enum_v` and then it provides the case for enums.

(We have attempted to follow this "primary is true" convention elsewhere, although there may be some inconsistent occurrences.)

Reply:
I'll push this stylistic change since I'm preparing a batch of MSVC-internal product and test changes.